msidoc.md

```diff
@@ -0,0 +1,415 @@
+94c3992a02ae: Preparing
+5ac11183120f: Preparing
+9abb510f612b: Preparing
+1dfbdf308b77: Preparing
+2ec940494cc0: Preparing
+6dfaec39e726: Preparing
+6dfaec39e726: Waiting
+1dfbdf308b77: Layer already exists
+2ec940494cc0: Layer already exists
+6dfaec39e726: Layer already exists
+5ac11183120f: Pushed
+94c3992a02ae: Pushed
+9abb510f612b: Pushed
+cf3: digest: sha256:16e796bb40d2b0f78102a3d17f69444db10a6862ef40a771fd069f7cab8652a3 size: 1577
+2019/05/16 21:53:54 Successfully pushed image: myregistry.azurecr.io/my-website:cf3
+2019/05/16 21:53:54 Executing step ID: acb_step_3. Working directory: 'ManagedIdentities', Network: 'acb_default_network'
+2019/05/16 21:53:54 Launching container with name: acb_step_3
+[
+  {
+    "environmentName": "AzureCloud",
+    "id": "84c559c6-30a0-417c-ba06-8a2253b388c3",
+    "isDefault": true,
+    "name": "ACR - TEST",
+    "state": "Enabled",
+    "tenantId": "72f988bf-86f1-41af-91ab-2d7cd011db47",
+    "user": {
+      "assignedIdentityInfo": "MSI",
+      "name": "systemAssignedIdentity",
+      "type": "servicePrincipal"
+    }
+  }
+]
+2019/05/16 21:53:58 Successfully executed container: acb_step_3
+2019/05/16 21:53:58 Executing step ID: acb_step_4. Working directory: 'ManagedIdentities', Network: 'acb_default_network'
+2019/05/16 21:53:58 Launching container with name: acb_step_4
+[
+  "cf2",
+  "cf3"
+]
+2019/05/16 21:54:02 Successfully executed container: acb_step_4
+2019/05/16 21:54:02 Step ID: acb_step_0 marked as successful (elapsed time in seconds: 1.070111)
+2019/05/16 21:54:02 Step ID: acb_step_1 marked as successful (elapsed time in seconds: 14.031206)
+2019/05/16 21:54:02 Step ID: acb_step_2 marked as successful (elapsed time in seconds: 7.053921)
+2019/05/16 21:54:02 Step ID: acb_step_3 marked as successful (elapsed time in seconds: 3.741742)
+2019/05/16 21:54:02 Step ID: acb_step_4 marked as successful (elapsed time in seconds: 4.274584)
+ 
+Run ID: cf3 was successful after 38s
+```
+
+## Private registry login using Managed Identities
+
+
+We will use System Identity for this example.
+
+Task:
+
+`testtask.yaml`
+``` yaml
+version: v1.0.0
+steps:
+  - build: -t {{.Values.REGISTRY1}}/hello-world:{{.Run.ID}} . -f hello-world.dockerfile
+  - push: ["{{.Values.REGISTRY1}}/hello-world:{{.Run.ID}}"]
+  - build: -t {{.Values.REGISTRY2}}/hello-world:{{.Run.ID}} . -f hello-world.dockerfile
+  - push: ["{{.Values.REGISTRY2}}/hello-world:{{.Run.ID}}"]
+```
+
+
+``` sh
+$ az acr task create -n multiple-reg -r myregistry -c https://github.com/Azure-Samples/acr-tasks.git#:multipleRegistries \
+   -f testtask.yaml --commit-trigger-enabled false --pull-request-trigger-enabled false --assign-identity
+
+{
+  "agentConfiguration": {
+    "cpu": 2
+  },
+  "creationDate": "2019-05-16T22:54:04.620200+00:00",
+  "credentials": null,
+  "id": "/subscriptions/84c559c6-30a0-417c-ba06-8a2253b388c3/resourceGroups/samashahtest-rg/providers/Microsoft.ContainerRegistry/registries/myregistry/tasks/multiple-reg",
```

Comment:
You might want to remove/anonymize identifying info in return

Reply:
whats the usual practice of anonymizing GUIDs?
There is `subscriptionId`, `principalId`, `tenantId` , etc. Should I anonymize them all with xxxxxxxxx-xxxxx-xxxx-xxxx-xxxxxxxxxxxxx ?

Reply:
@shahabhijeet - Yes, that works